<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>LoRaDemodGUI</class>
 <widget class="RollupWidget" name="LoRaDemodGUI">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>410</width>
    <height>620</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>380</width>
    <height>620</height>
   </size>
  </property>
  <property name="font">
   <font>
    <family>Liberation Sans</family>
    <pointsize>9</pointsize>
   </font>
  </property>
  <property name="windowTitle">
   <string>LoRa Demodulator</string>
  </property>
  <widget class="QWidget" name="settingsContainer" native="true">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>20</y>
     <width>390</width>
     <height>102</height>
    </rect>
   </property>
   <property name="minimumSize">
    <size>
     <width>0</width>
     <height>102</height>
    </size>
   </property>
   <property name="windowTitle">
    <string>RF/demod Settings</string>
   </property>
   <widget class="QLabel" name="bwLabel">
    <property name="geometry">
     <rect>
      <x>2</x>
      <y>50</y>
      <width>22</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>BW</string>
    </property>
   </widget>
   <widget class="QLabel" name="spreadLabel">
    <property name="geometry">
     <rect>
      <x>2</x>
      <y>70</y>
      <width>22</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>SF</string>
    </property>
   </widget>
   <widget class="QSlider" name="BW">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>50</y>
      <width>251</width>
      <height>16</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>Bandwidth</string>
    </property>
    <property name="minimum">
     <number>0</number>
    </property>
    <property name="maximum">
     <number>10</number>
    </property>
    <property name="pageStep">
     <number>1</number>
    </property>
    <property name="value">
     <number>5</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QSlider" name="Spread">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>70</y>
      <width>131</width>
      <height>16</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>Spreading factor</string>
    </property>
    <property name="minimum">
     <number>7</number>
    </property>
    <property name="maximum">
     <number>12</number>
    </property>
    <property name="pageStep">
     <number>1</number>
    </property>
    <property name="value">
     <number>10</number>
    </property>
    <property name="sliderPosition">
     <number>10</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QLabel" name="SpreadText">
    <property name="geometry">
     <rect>
      <x>170</x>
      <y>70</y>
      <width>30</width>
      <height>16</height>
     </rect>
    </property>
    <property name="minimumSize">
     <size>
      <width>30</width>
      <height>0</height>
     </size>
    </property>
    <property name="text">
     <string>10</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="BWText">
    <property name="geometry">
     <rect>
      <x>300</x>
      <y>50</y>
      <width>80</width>
      <height>16</height>
     </rect>
    </property>
    <property name="minimumSize">
     <size>
      <width>80</width>
      <height>0</height>
     </size>
    </property>
    <property name="text">
     <string>7813 Hz</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="deBitsLabel">
    <property name="geometry">
     <rect>
      <x>220</x>
      <y>70</y>
      <width>22</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>DE</string>
    </property>
   </widget>
   <widget class="QLabel" name="deBitsText">
    <property name="geometry">
     <rect>
      <x>350</x>
      <y>70</y>
      <width>30</width>
      <height>16</height>
     </rect>
    </property>
    <property name="minimumSize">
     <size>
      <width>30</width>
      <height>0</height>
     </size>
    </property>
    <property name="text">
     <string>0</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
    </property>
   </widget>
   <widget class="QSlider" name="deBits">
    <property name="geometry">
     <rect>
      <x>250</x>
      <y>70</y>
      <width>91</width>
      <height>16</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>Distance Enhancement bits i.e. log2 of number of FFT bins per effective sample</string>
    </property>
    <property name="minimum">
     <number>0</number>
    </property>
    <property name="maximum">
     <number>4</number>
    </property>
    <property name="pageStep">
     <number>1</number>
    </property>
    <property name="value">
     <number>0</number>
    </property>
    <property name="sliderPosition">
     <number>0</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>371</width>
      <height>26</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="deltaFrequencyLayout">
     <item>
      <widget class="QLabel" name="deltaFrequencyLabel">
       <property name="minimumSize">
        <size>
         <width>16</width>
         <height>0</height>
        </size>
       </property>
       <property name="text">
        <string>Df</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="ValueDialZ" name="deltaFrequency" native="true">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Maximum" vsizetype="Maximum">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="minimumSize">
        <size>
         <width>32</width>
         <height>16</height>
        </size>
       </property>
       <property name="font">
        <font>
         <family>Liberation Mono</family>
         <pointsize>12</pointsize>
        </font>
       </property>
       <property name="cursor">
        <cursorShape>PointingHandCursor</cursorShape>
       </property>
       <property name="focusPolicy">
        <enum>Qt::StrongFocus</enum>
       </property>
       <property name="toolTip">
        <string>Demod shift frequency from center in Hz</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="deltaUnits">
       <property name="text">
        <string>Hz </string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="Line" name="line">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QLabel" name="sLabel">
       <property name="text">
        <string>S</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="sText">
       <property name="minimumSize">
        <size>
         <width>30</width>
         <height>0</height>
        </size>
       </property>
       <property name="toolTip">
        <string>De-chirped signal level</string>
       </property>
       <property name="text">
        <string>-50.0</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="snrLabel">
       <property name="minimumSize">
        <size>
         <width>35</width>
         <height>0</height>
        </size>
       </property>
       <property name="text">
        <string>SNR</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="snrText">
       <property name="minimumSize">
        <size>
         <width>25</width>
         <height>0</height>
        </size>
       </property>
       <property name="toolTip">
        <string>De-chirped SNR level</string>
       </property>
       <property name="text">
        <string>10.0</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="sUnits">
       <property name="text">
        <string>dB</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QWidget" name="payloadContainer" native="true">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>130</y>
     <width>390</width>
     <height>160</height>
    </rect>
   </property>
   <property name="minimumSize">
    <size>
     <width>0</width>
     <height>160</height>
    </size>
   </property>
   <property name="windowTitle">
    <string>Payload</string>
   </property>
   <widget class="QLabel" name="messageLabel">
    <property name="geometry">
     <rect>
      <x>2</x>
      <y>40</y>
      <width>32</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Msg</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="hexText">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>120</y>
      <width>351</width>
      <height>20</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="hexLabel">
    <property name="geometry">
     <rect>
      <x>2</x>
      <y>120</y>
      <width>32</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Hex</string>
    </property>
   </widget>
   <widget class="QPlainTextEdit" name="messageText">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>40</y>
      <width>351</width>
      <height>75</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="schemeLabel">
    <property name="geometry">
     <rect>
      <x>2</x>
      <y>10</y>
      <width>50</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Scheme</string>
    </property>
   </widget>
   <widget class="QComboBox" name="scheme">
    <property name="geometry">
     <rect>
      <x>60</x>
      <y>8</y>
      <width>86</width>
      <height>20</height>
     </rect>
    </property>
    <item>
     <property name="text">
      <string>LoRa</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>ASCII</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>TTY</string>
     </property>
    </item>
   </widget>
   <widget class="QPushButton" name="clear">
    <property name="geometry">
     <rect>
      <x>4</x>
      <y>60</y>
      <width>20</width>
      <height>20</height>
     </rect>
    </property>
    <property name="maximumSize">
     <size>
      <width>24</width>
      <height>24</height>
     </size>
    </property>
    <property name="toolTip">
     <string>Clear text</string>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="icon">
     <iconset resource="../../../sdrgui/resources/res.qrc">
      <normaloff>:/sweep.png</normaloff>:/sweep.png</iconset>
    </property>
    <property name="autoDefault">
     <bool>false</bool>
    </property>
   </widget>
   <widget class="QToolButton" name="mute">
    <property name="geometry">
     <rect>
      <x>150</x>
      <y>8</y>
      <width>20</width>
      <height>20</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>Run/Stop decoder</string>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="icon">
     <iconset resource="../../../sdrgui/resources/res.qrc">
      <normaloff>:/stop.png</normaloff>
      <normalon>:/play.png</normalon>:/stop.png</iconset>
    </property>
    <property name="checkable">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QLabel" name="eomSquelchLabel">
    <property name="geometry">
     <rect>
      <x>180</x>
      <y>10</y>
      <width>35</width>
      <height>19</height>
     </rect>
    </property>
    <property name="text">
     <string>EOM</string>
    </property>
   </widget>
   <widget class="QDial" name="eomSquelch">
    <property name="geometry">
     <rect>
      <x>210</x>
      <y>8</y>
      <width>22</width>
      <height>22</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>End Of Message squelch factor</string>
    </property>
    <property name="minimum">
     <number>40</number>
    </property>
    <property name="maximum">
     <number>121</number>
    </property>
    <property name="pageStep">
     <number>1</number>
    </property>
    <property name="value">
     <number>60</number>
    </property>
   </widget>
   <widget class="QLabel" name="eomSquelchText">
    <property name="geometry">
     <rect>
      <x>230</x>
      <y>10</y>
      <width>28</width>
      <height>19</height>
     </rect>
    </property>
    <property name="text">
     <string>10.0</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="messageLengthLabel">
    <property name="geometry">
     <rect>
      <x>272</x>
      <y>10</y>
      <width>20</width>
      <height>19</height>
     </rect>
    </property>
    <property name="text">
     <string>ML</string>
    </property>
   </widget>
   <widget class="QDial" name="messageLength">
    <property name="geometry">
     <rect>
      <x>290</x>
      <y>8</y>
      <width>22</width>
      <height>22</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>Message (payload) length in number of symbols</string>
    </property>
    <property name="minimum">
     <number>20</number>
    </property>
    <property name="maximum">
     <number>255</number>
    </property>
    <property name="pageStep">
     <number>1</number>
    </property>
    <property name="value">
     <number>255</number>
    </property>
   </widget>
   <widget class="QLabel" name="messageLengthText">
    <property name="geometry">
     <rect>
      <x>310</x>
      <y>10</y>
      <width>25</width>
      <height>19</height>
     </rect>
    </property>
    <property name="text">
     <string>255</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
    </property>
   </widget>
   <widget class="QLineEdit" name="syncWord">
    <property name="geometry">
     <rect>
      <x>350</x>
      <y>9</y>
      <width>25</width>
      <height>20</height>
     </rect>
    </property>
    <property name="focusPolicy">
     <enum>Qt::ClickFocus</enum>
    </property>
    <property name="toolTip">
     <string>Sync word (1 byte hex)</string>
    </property>
    <property name="inputMask">
     <string>HH</string>
    </property>
    <property name="text">
     <string>00</string>
    </property>
   </widget>
  </widget>
  <widget class="QWidget" name="spectrumContainer" native="true">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>300</y>
     <width>390</width>
     <height>310</height>
    </rect>
   </property>
   <property name="minimumSize">
    <size>
     <width>373</width>
     <height>0</height>
    </size>
   </property>
   <property name="windowTitle">
    <string>De-chirped Spectrum</string>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout">
    <property name="spacing">
     <number>2</number>
    </property>
    <property name="leftMargin">
     <number>3</number>
    </property>
    <property name="topMargin">
     <number>3</number>
    </property>
    <property name="rightMargin">
     <number>3</number>
    </property>
    <property name="bottomMargin">
     <number>3</number>
    </property>
    <item>
     <widget class="GLSpectrum" name="glSpectrum" native="true">
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>300</height>
       </size>
      </property>
     </widget>
    </item>
    <item>
     <widget class="GLSpectrumGUI" name="spectrumGUI" native="true"/>
    </item>
   </layout>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>RollupWidget</class>
   <extends>QWidget</extends>
   <header>gui/rollupwidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>GLSpectrum</class>
   <extends>QWidget</extends>
   <header>gui/glspectrum.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>GLSpectrumGUI</class>
   <extends>QWidget</extends>
   <header>gui/glspectrumgui.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>ValueDialZ</class>
   <extends>QWidget</extends>
   <header>gui/valuedialz.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../../../sdrgui/resources/res.qrc"/>
 </resources>
 <connections/>
</ui>
